tests/test_cli.py

```diff
@@ -0,0 +1,264 @@
+        ("sib21", im.Sibyll21),
+        ("sibyll-2.3", im.Sibyll23),
+    ),
+)
+def test_model_1(spec, Model):
+    run(
+        "-S",
+        "100",
+        "-s",
+        "1",
+        "-m",
+        spec,
+        stdout=f"Model[ \t]*{Model.label}",
+        file=f"impy_{Model.pyname.lower()}_1_2212_2212_100.hepmc",
+    )
+
+
+@pytest.mark.skipif(
+    platform.system() == "Windows", reason="Pythia-8 not available on Windows"
+)
+def test_model_2():
+    run(
+        "-m",
+        "py8",
+        returncode=1,
+        stderr="Error: model py8 is ambiguous, matches Pythia-6.428, Pythia-8.307",
+    )
+
+
+def test_model_3():
+    run(
+        "-m",
+        "sib",
+        returncode=1,
+        stderr=(
+            "Error: model sib is ambiguous, matches SIBYLL-2.1, SIBYLL-2.3, "
+            "SIBYLL-2.3c, SIBYLL-2.3d"
+        ),
+    )
+
+
+@pytest.mark.parametrize(
+    "spec,expected",
+    (
+        ("pi+", 211),
+        ("pi-", -211),
+        ("p", 2212),
+        ("O", 1000080160),
+        ("211", 211),
+        ("2212", 2212),
+    ),
+)
+def test_projectile(spec, expected):
+    p = Particle.from_pdgid(expected)
+    name = p.name.replace("+", "\\+")
+    run(
+        "-S",
+        "100",
+        "-s",
+        "1",
+        "-i",
+        spec,
+        stdout=f"Projectile[ \t]*{name} \\({expected}\\)",
+        file=f"impy_eposlhc_1_{expected}_2212_100.hepmc",
+    )
+
+
+@pytest.mark.parametrize(
+    "spec,expected",
+    (
+        ("p", 2212),
+        ("C", 1000060120),
+        ("O", 1000080160),
+        ("2212", 2212),
+    ),
+)
+def test_target(spec, expected):
+    p = Particle.from_pdgid(expected)
+    run(
+        "-S",
+        "100",
+        "-s",
+        "1",
+        "-I",
+        spec,
+        stdout=f"Target[ \t]*{p.name} \\({expected}\\)",
+        file=f"impy_eposlhc_1_2212_{expected}_100.hepmc",
+    )
+
+
+def test_momentum_1():
+    run(
+        "-s",
+        "1",
+        "-p 1000",
+        "-P -1000",
+        stdout="sqrt\\(s\\)[ \t]*2000 GeV",
+        file="impy_eposlhc_1_2212_2212_2000.hepmc",
+    )
+
+
+def test_momentum_2():
+    run(
+        "-s",
+        "1",
+        "-p 1000",
+        "-P 0",
+        stdout="""\
+[ |]*Projectile momentum[ \t]*1000 GeV/c *|
+[ |]*Target momentum[ \t]*0 GeV/c *|
+[ |]*sqrt\\(s\\)[ \t]*43.3394 GeV\
+""",
+        file="impy_eposlhc_1_2212_2212_43.hepmc",
+    )
+
+
+def test_format_1():
+    run(
+        "-s",
+        "1",
+        "-S",
+        "100",
+        stdout="Format[ \t]*hepmc",
+        file="impy_eposlhc_1_2212_2212_100.hepmc",
+    )
+
+
+@pytest.mark.parametrize("format", ("hepmc", "hepmcgz", "root"))
+def test_format_2(format):
+    ext = format
+    if ext.endswith("gz"):
+        ext = ext[:-2] + ".gz"
+
+    run(
+        "-s",
+        "1",
+        "-S",
+        "100",
+        "-o",
+        format,
+        stdout=f"Format[ \t]*{format}",
+        file=f"impy_eposlhc_1_2212_2212_100.{ext}",
+    )
```

Comment:
I get an error " undefined symbol: Npy__PyLong_AsInt" for most of the tests. What could be  the reason here?

Reply:
When and where? I did not see these and they are not listed in the CI runs. Perhaps your build is corrupted and you need to rebuild from scratch. This undefined symbol indicates that libpython is not linked properly into impy on your machine. This cannot have anything to do with this PR, it is a build system issue.

Reply:
This was a problem with pyf file from Epos. After regeneration of the pyf file tests pass.

Reply:
So it can be resolved?